packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -77,8 +77,16 @@ pub fn generate_script_main_function(
         ));
     }
 
-    let args = function_arguments(main_function_abi, types)?;
+    let output_type = resolve_fn_output_type(main_function_abi, types)?;
 
+    let args = function_arguments(main_function_abi, types)?;
+    if args.iter().filter(|c| c.field_type.uses_vectors()).count() != 0 {
+        return Err(Error::CompilationError(
+            "Script main function contains a vector in its argument types. This currently isn't \
```

Comment:
Vectors as contract method inputs are supported. The reason they might not be for scripts is the `resolve()` business if I'm not mistaken. Haven't checked but we should make a task for it.

In any case, returning vectors isn't supported anywhere, so you should add a
`output_type.uses_vectors` guard somewhere.

Reply:
The guard for outputs is already contained in `resolve_fn_output_types` :)
Created issue #655.